launcher/firstLaunch/firstlaunch_moc.cpp

```diff
@@ -341,7 +341,13 @@ void FirstLaunchView::extractGogData()
 
 			o.preserve_file_times = true; // also correctly closes file -> without it: on Windows the files are not written completly
 
-			process_file(tmpFileExe.toStdString(), o);
+			this->setEnabled(false);
+			process_file(tmpFileExe.toStdString(), o, [this](float percent) {
+				ui->pushButtonGogInstall->setText(tr("Installing...") + " " + QString::number(static_cast<int>(percent * 100)) + "%");
```

Comment:
Please check, not sure how to show '%' sign in Qt formatting. You might need '%%' instead
```suggestion
			process_file(tmpFileExe.toStdString(), o, [this](float progress) {
				ui->pushButtonGogInstall->setText(tr("Installing... %1%").arg(progress* 100));
```

Oh, and your variable is not 'percent', this is progress or ratio. Percentages should go from 0 to 100, but this parameter goes from 0 to 1, so should be named appropriately

Reply:
don't think you need %%

is extracting done on the main thread? I think it's better to add `qApp->processEvents()` to make sure that UI updates properly. If not on main thread, then you need to use `ui->pushButtonGogInstall->invoke()` iirc